### PR TITLE
[11.0][IMP] Disable password expiration as in version 12.0 (backport)

### PR DESCRIPTION
**Pull request by javiermatos**
Implements fix for https://github.com/OCA/server-auth/issues/100

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Why aren't you cherry-picking the commit with the change? And I see that you are also doing some unneeded changes like introducing spaces in some docstrings, which you shouldn't (and more as that's not the way to put them).

**Comment by javiermatos** [comment]
Thank you for the feedback @pedrobaeza. I have never used the cherry-picking feature from git (shame on me...). Anyway, I have located it here https://github.com/OCA/server-auth/commit/d683bf71f72e831a5b3af67fec6ba881291ca532 but I do not think we can just apply it as it introduces many changes for migrating to 12.0 that are not valid in our case. Am I wrong with that?

Regarding the spaces in docstrings, I introduced them as those spaces at the beginning and at the end of the docstrings are present in all the other docstrings for the tests of that module. I though the one that initially wrote the tests just forgot to add them.

I can revert the unneeded changes if you prefer that and just focus on the specific changes that are required.

(I have signed the CLA, so it is ready)

**Comment by javiermatos** [comment]
Could we please integrate this into the 11.0 branch? I think everything is as it should, but if not, just tell me what should be fixed.

**Review by oihane — APPROVED** [review]
LGTM

File: password_security/models/res_users.py [hunk 1]
```diff
@@ -97,6 +97,10 @@ def _password_has_expired(self):
         self.ensure_one()
         if not self.password_write_date:
             return True
+
+        if not self.company_id.password_expiration:
```
Comment (pedrobaeza) [inline password_security/models/res_users.py:101]: Where is this field defined?

Comment (javiermatos) [reply password_security/models/res_users.py:101]: The field is defined in password_security/models/res_company.py

```
class ResCompany(models.Model):
    _inherit = 'res.company'

    password_expiration = fields.Integer(
        'Days',
        default=60,
        help='How many days until passwords expire',
    )
    ...
```

Comment (pedrobaeza) [reply password_security/models/res_users.py:101]: OK, seen. I have mixed modules.

**Review by pedrobaeza — APPROVED** [review]

File: password_security/models/res_users.py (hunk 1 above)
Comment (javiermatos) [reply password_security/models/res_users.py:101]: Great! Thank you so much! :)

**Pull request merged**